src/ophyd_async/core/mock_signal_utils.py

```diff
@@ -0,0 +1,183 @@
+import asyncio
+import itertools
+from contextlib import asynccontextmanager, contextmanager
+from typing import Any, Callable, Dict, Generator, Iterator, List
+from unittest.mock import ANY
+
+from ophyd_async.core.signal import Signal
+from ophyd_async.core.utils import ReadingValueCallback, T
+
+from .mock_signal_backend import MockSignalBackend
+
+
+def _get_mock_signal_backend(signal: Signal) -> MockSignalBackend:
+    assert isinstance(signal._backend, MockSignalBackend), (
+        "Expected to receive a `MockSignalBackend`, instead "
+        f" received {type(signal._backend)}. "
+    )
+    return signal._backend
+
+
+def set_mock_value(signal: Signal[T], value: T):
+    """Set the value of a signal that is in mock mode."""
+    backend = _get_mock_signal_backend(signal)
+    backend._set_value(value)
+
+
+def set_mock_put_proceeds(signal: Signal[T], proceeds: bool):
+    """Allow or block a put with wait=True from proceeding"""
+    backend = _get_mock_signal_backend(signal)
+
+    if proceeds:
+        backend.put_proceeds.set()
+    else:
+        backend.put_proceeds.clear()
+
+
+def set_mock_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> None:
```

Comment:
Deletion and name changes as per https://github.com/bluesky/ophyd-async/issues/94

Reply:
https://github.com/bluesky/ophyd-async/issues/94#issuecomment-2088428517